core/src/test/java/org/infinispan/partitionhandling/PreferConsistencyStrategyTest.java

```diff
@@ -45,5 +77,49 @@ public void testAvoidingNullPointerExceptionWhenUpdatingPartitionWithNullTopolog
       Assert.assertEquals(RebalancingStatus.COMPLETE, status.getRebalancingStatus());
    }
 
+   public void testCoordinatorHasNewerTopology() throws Exception {
+      Address myAddress = new TestAddress(1);
+      Address otherAddress = new TestAddress(2);
 
+      manager.addPersistentAddressMapping(myAddress, PersistentUUID.randomUUID());
+      manager.addPersistentAddressMapping(otherAddress, PersistentUUID.randomUUID());
+
+      ClusterCacheStatus coordinatorStatus = new ClusterCacheStatus("does-not-matter", preferConsistencyStrategy, topologyManager, null, Optional.empty(), manager);
+      coordinatorStatus.doJoin(myAddress, createJoinInfo(myAddress));
+      coordinatorStatus.doJoin(otherAddress, createJoinInfo(otherAddress));
+      // at this stage current topology ID from CacheStatus is 2
+
+      // The cache response contains non-null stable topology but null current topology
+      CacheStatusResponse responseFromOtherNode = new CacheStatusResponse(null,
+            null,
+            createCacheTopology(1, myAddress, otherAddress),
+            AvailabilityMode.AVAILABLE);
+
+      //when
+      preferConsistencyStrategy.onPartitionMerge(coordinatorStatus, Arrays.asList(responseFromOtherNode));
```

Comment:
TBH I'm likely the one at fault, because `PreferConsistencyStrategy` and `ClusterCacheStatus` have way too little error checking, and there's no way to know what inputs are acceptable when a cache is in a certain state. E.g. it shouldn't be possible for a cache status response to have a `null` current topology and a non-null stable topology, because a node should only send a `null` current topology if it didn't receive the join response yet.